core/src/main/scala/filodb.core/memstore/TimeSeriesShard.scala

```diff
@@ -1068,7 +1053,15 @@ class TimeSeriesShard(val dataset: Dataset,
     try {
       val part: FiloPartition = getOrAddPartition(recordBase, recordOff, group, ingestOffset)
       if (part == OutOfMemPartition) { disableAddPartitions() }
-      else { part.asInstanceOf[TimeSeriesPartition].ingest(binRecordReader, overflowBlockFactory) }
+      else {
+        part.asInstanceOf[TimeSeriesPartition].ingest(binRecordReader, overflowBlockFactory)
+        if (!activelyIngesting.get(part.partID)) {
+          // time series was inactive and has just started re-ingesting
+          updatePartEndTimeInIndex(part.asInstanceOf[TimeSeriesPartition], Long.MaxValue)
+          timeBucketBitmaps.get(currentIndexTimeBucket).set(part.partID)
+          activelyIngesting.set(part.partID)
```

Comment:
Need to hold the lock here too, over the two calls. Or an atomic get/set method: insert? setIfNotExists?